libtest/comparison.hpp

```diff
@@ -93,7 +93,7 @@ bool _compare(const char *file, int line, const char *func, const T1_comparable&
 template <class T1_comparable, class T2_comparable>
 bool _compare_strcmp(const char *file, int line, const char *func, const T1_comparable& __expected, const T2_comparable& __actual)
 {
-  if (__expected == NULL)
+  /*if (__expected == NULL)
```

Comment:
If the code is not to be compiled, and does not document the code around it, then it should be removed, not commented out.

Reply:
Thanks for your edits. You are right. I wanted to comment that out unsure about your opinion and if we wanted to fix this temporarily for users compiling gearmand with next generation compilers (like gcc 6.3). 

But since you expressed your opinion, thanks. I am removin that line.

I will remove nbproject for now, but are you sure it should not be in gitignore? I've considered it a common practice, since many of IDEs does support including their project files in .gitignore.

If we would like to avoid polluting projects, I think we should consider `t/` and `tests/` directories to have their own `.gitignore` files. I am curious about your opinion on that matter.